moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1394,25 +1402,28 @@ bool TrajectoryExecutionManager::executePart(std::size_t part_index)
     }
 
     // compute the expected duration of the trajectory and find the part of the trajectory that takes longest to execute
-    ros::Time current_time = ros::Time::now();
-    ros::Duration expected_trajectory_duration(0.0);
+
+    rclcpp::Time current_time = rclcpp::Clock().now();
+    rclcpp::Duration expected_trajectory_duration(0.0);
     int longest_part = -1;
     for (std::size_t i = 0; i < context.trajectory_parts_.size(); ++i)
     {
-      ros::Duration d(0.0);
+      rclcpp::Duration d(0.0);
       if (!(context.trajectory_parts_[i].joint_trajectory.points.empty() &&
             context.trajectory_parts_[i].multi_dof_joint_trajectory.points.empty()))
       {
-        if (context.trajectory_parts_[i].joint_trajectory.header.stamp > current_time)
-          d = context.trajectory_parts_[i].joint_trajectory.header.stamp - current_time;
-        if (context.trajectory_parts_[i].multi_dof_joint_trajectory.header.stamp > current_time)
-          d = std::max(d, context.trajectory_parts_[i].multi_dof_joint_trajectory.header.stamp - current_time);
+        if (rclcpp::Time(context.trajectory_parts_[i].joint_trajectory.header.stamp) > current_time)
```

Comment:
Isn't the `rclcpp::Time` constructor redundant here? I thought the stamp (builtin_interfaces/Time) directly converts to rclcpp::Time anyway.

Reply:
I think `rclcpp::Time` constructor is still necessary here, because I only found conversion operator that converts `rclcpp::Time` to `builtin_interfaces::msg::Time`, and not vice versa.